### PR TITLE
Added support for GitHub Actions CI

### LINKED ISSUE #1670: Add support for GitHub Actions CI
### Bug Report
Add support for GitHub Actions CI to the repository. This brings down the run time for all the tests to 12-13 mins (assuming dependencies are cached) from 15-20 mins with Travis CI. Carthage dependencies are cached and only rebuilt if needed.
 
There are three separate jobs created:
1. The Swift example app is built.
1. The Obj-C example app is built.
1. The unit tests are built and run.

### PR DESCRIPTION
**Pull request by NicoleYarroch**
Fixes #1670 

This PR is **ready** for review.

### Risk
This PR makes **no** API changes. 

### Testing Plan
- [ ] I have verified that I have not introduced new warnings in this PR (or explain why below)
- [ ] I have run the unit tests with this PR
- [ ] I have tested this PR against Core and verified behavior (if applicable, if not applicable, explain why below).

The library was not modified. Only support for GitHub Actions CI was added.

#### Unit Tests
The library was not modified. Only support for GitHub Actions CI was added.

#### Core Tests
The library was not modified. Only support for GitHub Actions CI was added.

Core version / branch / commit hash / module tested against: n/a
HMI name / version / branch / commit hash / module tested against: n/a

### Summary
Add support for GitHub Actions CI to the repository and removed the Travis CI config file. This brings down the run time for all the tests to 12-13 mins (assuming dependencies are cached) from 15-20 mins with Travis CI. Carthage dependencies are cached and only rebuilt if needed. 
 
There are three separate jobs created:
1. The Swift example app is built.
1. The Obj-C example app is built.
1. The unit tests are built and run.

The workflow jobs will be triggered on a pull-request or a push. The result for each job can be viewed at the bottom of the "Conversation" tab of the pull request (Click on "Show all Checks"). 

##### Enhancements
Add support for GitHub Actions CI to the repository.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
+name: SmartDeviceLink Unit Tests
+
+# This workflow is triggered on a push or pull request.
+on:
+    push:
+      branches:
+        - '*'
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply .github/workflows/test.yml:7]: Can you explain what you think is happening? This is a [workaround](https://github.community/t/how-to-trigger-an-action-on-push-or-pull-request-but-not-both/16662/3) so two workflows are not started when a push is made to a pull request (i.e. one for the push and one for the pull). If you look at the Java-Suite repo they do not have this workaround so they generate a two workflows.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      tags-ignore: # Don't run workflow when a tag is created
+        - v*
+    pull_request:
+      branches:
+        - '*'
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply .github/workflows/test.yml]: Fixed to use 4 spaces as a indent

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -1,97 +1,83 @@
...
+name: SmartDeviceLink Tests
 
 # This workflow is triggered on a push or pull request.
 on:
+    # Runs the workflow when someone pushes to a repository branch, which triggers the push event.
     push:
-      branches:
-        - '*'
-      tags-ignore: # Don't run workflow when a tag is created
-        - v*
```
Comment (joeljfischer) [inline .github/workflows/test.yml]: Should these have been removed? Now there's nothing underneath it, which looks odd at least.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,83 @@
+name: SmartDeviceLink Tests
+
+# This workflow is triggered on a push or pull request.
+on:
+    # Runs the workflow when someone pushes to a repository branch, which triggers the push event.
+    push:
+
+    # Runs the workflow only when someone opens or reopens a pull request (there are a bunch of activity types that can trigger a workflow that we don't want - like labeling, assigning, and locking the PR). This prevents two workflows from being triggered when a push is made to a pull request branch because the pull request activity type of "synchronize" also triggers another, separate, workflow.
+    pull_request:
+        types: [opened, reopened]
```
Comment (joeljfischer) [inline .github/workflows/test.yml:10]: I don't think we need `reopened`. If you open a CI build by hitting "details", you can re-run the job.

File: .github/workflows/test.yml (hunk 3 above)
Comment (joeljfischer) [reply .github/workflows/test.yml]: At least add a comment to explain that nothing needs to be there so it runs on every push if that's what's happening

File: README.md [hunk 5]
```diff
@@ -2,7 +2,7 @@
 [![Carthage compatible](https://img.shields.io/badge/Carthage-compatible-4BC51D.svg?style=flat)](https://github.com/Carthage/Carthage)
 [![Version](https://img.shields.io/cocoapods/v/SmartDeviceLink-iOS.svg?style=flat)](https://cocoapods.org/pods/SmartDeviceLink-iOS)
 [![License](https://img.shields.io/cocoapods/l/SmartDeviceLink-iOS.svg?style=flat)](https://cocoapods.org/pods/SmartDeviceLink-iOS)
-[![Build Status](https://img.shields.io/travis/smartdevicelink/sdl_ios/master.svg?style=flat)](https://travis-ci.org/smartdevicelink/sdl_ios)
+![SmartDeviceLink Tests](https://github.com/smartdevicelink/sdl_ios/workflows/SmartDeviceLink%20Tests/badge.svg)
```
Comment (NicoleYarroch) [inline README.md:5]: I removed the Travis badge and replaced it with the workflows test badge.

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -0,0 +1,82 @@
+name: SmartDeviceLink Tests
+
+# This workflow is triggered on a push or pull request.
+on:
+    # Runs the workflow when someone pushes to any repository branch.
+    push:
```
Comment (NicoleYarroch) [inline .github/workflows/test.yml:6]: I can't comment on your comment for some reason so I'll just leave a comment here...

I can't figure out the formatting that allows you to remove the colon after the keyword is you have another keyword that does need the colon, so I am just going to leave it as is. I added a comment to the "pull_request" keyword explaining how it works.

File: .github/workflows/test.yml (hunk 3 above)
Comment (NicoleYarroch) [reply .github/workflows/test.yml]: I moved the comment so hopefully it is more clear.

File: .github/workflows/test.yml (hunk 4 above)
Comment (NicoleYarroch) [reply .github/workflows/test.yml]: Removed

**Pull request merged**